### PR TITLE
Adding rook-recovery role which can be used for deploying the rook je…

### PR DESCRIPTION
**Pull request by akraitman**
…nkins instance in the rook CI aws vpc, The deployied instance will have two ebs volumes were the jobs and the jobs configuration data is located

Signed-off-by: Adam Kraitman <akraitma@redhat.com>

### CODE REVIEW

File: roles/rook-recovery/README.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+rook-recovery
+==========
+
+This role is used to deploy the rook jenkins instance in the rook CI aws vpc, The deployied instance will have two ebs volumes were the jobs and the jobs configuration data is located
```
Comment (djgalloway) [inline roles/rook-recovery/README.rst]: Typo

File: roles/rook-recovery/README.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+rook-recovery
+==========
+
+This role is used to deploy the rook jenkins instance in the rook CI aws vpc, The deployied instance will have two ebs volumes were the jobs and the jobs configuration data is located 
+
+Notes
++++++
+In order to make him avalible to public network once the instance is deployed you will need to add him to the load balancing target group under the rook ci vpc
```
Comment (djgalloway) [inline roles/rook-recovery/README.rst]: Typo

File: roles/rook-recovery/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
+  - name: Include secrets
+    include_vars: "{{ secrets_path | mandatory }}/aws.yaml"
+    no_log: true
+    tags:
+      - always
+
+  - name: Launch instance
+    ec2:
+       aws_access_key: "{{ aws_access_key }}"
```
Comment (djgalloway) [inline roles/rook-recovery/tasks/main.yml]: Looks like there's one extra space under `ec2:` than there should be.

File: roles/rook-recovery/vars/main.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+instance_type: m4.large
+security_group: rook-jenkins-group
+image: ami-09c003703cafa9056
+region: us-east-1
+vpc_subnet_id: subnet-c72b609b
+instance_name: Recovery-Rook-Jenkins
+aws_tags:
+  Name: "{{ instance_name }}"
+  Application: "Jenkins"
+
```
Comment (djgalloway) [inline roles/rook-recovery/vars/main.yml]: Extra `\n`

**Comment by djgalloway** [comment]
What is the source of the AMI?  Is it a generic Jenkins instance or is it a backup of the existing Jenkins instance?  Something else?

**Comment by akraitman** [comment]
> What is the source of the AMI? Is it a generic Jenkins instance or is it a backup of the existing Jenkins instance? Something else?

It's stable backup of a Jenkins instance because we need the ebs volumes were the jobs and configuration data is located

**Comment by djgalloway** [comment]
How feasible is it to have this role repopulate a plain/vanilla Jenkins instance using the existing Jenkinsfile from https://github.com/rook/rook/blob/master/Jenkinsfile ?

File: roles/rook-recovery/README.rst (hunk 1 above)
Comment (akraitman) [reply roles/rook-recovery/README.rst]: fixed

File: roles/rook-recovery/README.rst (hunk 2 above)
Comment (akraitman) [reply roles/rook-recovery/README.rst]: fixed

File: roles/rook-recovery/tasks/main.yml (hunk 3 above)
Comment (akraitman) [reply roles/rook-recovery/tasks/main.yml]: fixed

File: roles/rook-recovery/vars/main.yml (hunk 4 above)
Comment (akraitman) [reply roles/rook-recovery/vars/main.yml]: fixed

**Comment by akraitman** [comment]
> How feasible is it to have this role repopulate a plain/vanilla Jenkins instance using the existing Jenkinsfile from https://github.com/rook/rook/blob/master/Jenkinsfile ?

It's something i will need to investigate and asses because that would need more steps that would connect and mount the ebs volumes to that vanila instance before starting the jenkins container.
In any case the role would need to use the backup were the plugins and configuration data is located    so i am not sure if we will get a benefit in repopulating a plain/vanilla Jenkins instance over the way that the role is doing it now

File: roles/rook-recovery/README.rst (hunk 1 above)
Comment (djgalloway) [reply roles/rook-recovery/README.rst]: You need to `git add` all the files you fixed then `git commit --amend --no-edit` then `git push -f origin Rook-Recovery-Role` to push the fixes.

Comment (akraitman) [reply roles/rook-recovery/README.rst]: I did it yesterday but i forgot doing git add, now it should be fine

File: roles/rook-recovery/README.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+rook-recovery
+==========
+
+This role is used to deploy the rook jenkins instance in the rook CI aws vpc, The deployed instance will have two ebs volumes were the jobs and the jobs configuration data is located
```
Comment (djgalloway) [inline roles/rook-recovery/README.rst]: s/were/where

Comment (akraitman) [reply roles/rook-recovery/README.rst]: Fixed, thank you

**Pull request merged**